### PR TITLE
CLOUDP-246611: Wire context in the integration tests

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by s-urbaniak — APPROVED** [review]
This is outstanding! 👏 🎉 🙇

**Comment by s-urbaniak** [comment]
One comment. As per https://onsi.github.io/ginkgo/ we can directly wire the context in all ginkgo specs transparently, i.e.:

```
It("can save books", func(ctx SpecContext) {
      book := &books.Book{
        Title:  "Les Miserables",
        Author: "Victor Hugo",
        Pages:  2783,
      }

      Expect(libraryClient.SaveBook(ctx, book)).To(Succeed())
      Expect(libraryClient.ListBooks(ctx)).To(ContainElement(book))
})
```

> when such a node is detected Ginkgo will automatically supply a SpecContext object. This SpecContext object satisfies the context.Context interface and can be used anywhere a context.Context object is used. When a spec times out or is interrupted by the user (see below) Ginkgo will cancel the SpecContext to signal to the spec that it is time to exit. In the case above, it is assumed that libraryClient knows how to return once ctx is cancelled.

File: test/int/integration_suite_test.go [hunk 1]
```diff
@@ -71,27 +72,23 @@ const (
...
 
 	// These variables are initialized once per each node
+	ctx         context.Context
 	k8sClient   client.Client
 	atlasClient *admin.APIClient
 
 	// These variables are per each test and are changed by each BeforeRun
-	namespace         corev1.Namespace
-	cfg               *rest.Config
+	testNamespace     corev1.Namespace
```
Comment (josvazg) [inline test/int/integration_suite_test.go:83]: Would this not be a concurrency issue. I do not think the namespace can be shared across concurrent tests.

File: test/int/integration_suite_test.go [hunk 2]
```diff
@@ -71,27 +72,23 @@ const (
 var (
 	// This variable is "global" - as is visible only on the first ginkgo node
 	testEnv *envtest.Environment
+	cfg     *rest.Config
```
Comment (josvazg) [inline test/int/integration_suite_test.go:75]: What is this config about? Is this config immutable and the same for all tests, even concurrent ones?

File: test/int/integration_suite_test.go [hunk 3]
```diff
@@ -183,159 +197,162 @@ func defaultTimeouts() {
...
+			ObjectDeletionProtection:    deletionProtection,
+			SubObjectDeletionProtection: false,
+		}).SetupWithManager(k8sManager)
 		Expect(err).ToNot(HaveOccurred())
-	}()
+	})
 
-	return &namespace, managerCancelFunc
-}
+	By("Setting up database user controller", func() {
```
Comment (josvazg) [inline test/int/integration_suite_test.go:279]: nit

I would not have refactor these with such care. All this code needs to die soon.

There should be a single `setupAKOManager` under `pkg/` that can be called from **main**, **integration**, **integration wide** and **e2e** tests all the same.

hopefully next time we touch here is to remove code.

**Review by josvazg — COMMENTED** [review]
Just have a few questions, probably nothing, but would like to understand before I approve

**Comment by josvazg** [comment]
Oops! I just checked the CI expecting to see the usual e2e suspects there red, but the most of integration tests are red now. I believe this change introduces a problem.

**Comment by helderjs** [comment]
The task was done in another [PR](https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1611) keeping the context defined in the ticket.

**Pull request closed without merging**